taskiq/result.py

```diff
@@ -1,70 +1,12 @@
-import json
-import pickle  # noqa: S403
-from functools import partial
-from typing import Any, Callable, Dict, Generic, Optional, TypeVar
+# flake8: noqa
+from taskiq.compat import pydantic_version
 
-from pydantic import Field, validator
-from pydantic.generics import GenericModel
-from typing_extensions import Self
+if pydantic_version >= (2, 0):
```

Comment:
To make it even smoother and more beautiful, I would suggest the following. 

Since we're still going to use `from taskiq.result import Result`, we can structure package as this:


```
📁 taskiq
└── 📁 result
    ├── 📄 __init__.py
    ├── 📄 v1.py
    └── 📄 v2.py
```

in `__init__` we can do this:

```python
if pydantic_version >= Version('2.0'):
    from .v1 import TaskiqResult
else:
    from .v2 import TaskiqResult

__all__ = ["TaskiqResult"]

```

So, for users it's going to be easy and transparent as it is now. 

`from taskiq.result import TaskiqResult` will work as we want, and we will have a better structure of a project.

What do you think?

Reply:
much better